### PR TITLE
Fix time display emails

### PR DESCRIPTION
**Pull request by Calvinp**
In pause emails, when an expiration time was set:
- The timestamp displayed in a confusing way (24h clock always UTC). Now the format and timezone are configurable, and it defaults to 12h clock (but still UTC).
- A space was missing after the timestamp. This is fixed.

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java [hunk 1]
```diff
@@ -38,9 +38,13 @@
 
   private final Optional<String> uiBaseUrl;
   private final Optional<SMTPConfiguration> smtpConfiguration;
+  private String taskDatePattern;
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java]: make this final

File: SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java [hunk 2]
```diff
@@ -76,6 +77,8 @@
   @Min(value = 1, message = "Must be positive and non-zero")
   private int defaultBounceExpirationMinutes = 60;
 
+  private String defaultDatePattern = "MMM dd h:mm:ss a zzz";
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: If these config properties are going to be used for mail only, they should indicate in the name.

Comment (Calvinp) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: I'm not sure if they're going to be used for mail only or not - it's plausible that we would want to use them for something else. I don't know what that thing would be though.

File: SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java (hunk 1 above)
Comment (Calvinp) [inline SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java]: Done

File: SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java [hunk 3]
```diff
@@ -38,9 +38,13 @@
 
   private final Optional<String> uiBaseUrl;
   private final Optional<SMTPConfiguration> smtpConfiguration;
+  private final String taskDatePattern;
+  private final TimeZone timeZone;
 
   @Inject
   public MailTemplateHelpers(SandboxManager sandboxManager, SingularityConfiguration singularityConfiguration) {
+    taskDatePattern = singularityConfiguration.getDefaultDatePattern();
+    timeZone = singularityConfiguration.getTimeZone();
```
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java]: lets follow the format of the ones below with this.x = ...

File: SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java (hunk 2 above)
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: I think we should still make it more specific. At the moment, it is only used for mail, so let's make the name reflect that. Also it isn't really a default that we are falling back on, it is just a setting that will be used by everything. So maybe just calling it `mailerDatePattern` ?

File: SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java [hunk 4]
```diff
@@ -185,6 +188,8 @@
 
   private long startNewReconcileEverySeconds = TimeUnit.MINUTES.toSeconds(10);
 
+  private TimeZone timeZone = TimeZone.getTimeZone("UTC");
```
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: Will this initialize correctly from a String? (i.e. reading it from a yaml field set to 'EST' or 'UTC') It very well might, but something we should probably test to be sure

Comment (Calvinp) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: Yes, it is still able to start up.
I renamed the field to mailerTimeZone, and putting a top level line 'mailerTimeZone: UTC' into the yaml doesn't break the build.

File: SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java [hunk 5]
```diff
@@ -76,6 +77,8 @@
   @Min(value = 1, message = "Must be positive and non-zero")
   private int defaultBounceExpirationMinutes = 60;
 
+  private String mailerDatePattern = "MMM dd h:mm:ss a zzz";
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: we should move these mailer* config fields into `SMTPConfiguration`

File: SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java [hunk 6]
```diff
@@ -3,9 +3,11 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.TimeZone;
 
 import org.apache.commons.lang3.text.WordUtils;
 import org.apache.commons.lang3.time.DateFormatUtils;
+import org.apache.commons.lang3.tuple.Pair;
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java]: i dont think this import is necessary

**Pull request merged**